python/sglang/srt/mem_cache/memory_pool.py

```diff
@@ -861,33 +866,28 @@ def assign_flat_data(self, indices, flat_data):
         self.kv_buffer[:, :, indices] = flat_data
 
     def write_page_all_layers(self, host_indices, device_indices, device_pool):
-        device_indices_cpu = device_indices[:: self.page_size].cpu()
-        for i in range(len(device_indices_cpu)):
-            h_index = host_indices[i * self.page_size]
-            d_index = device_indices_cpu[i]
-            for j in range(self.layer_num):
-                self.kv_buffer[0, j, h_index : h_index + self.page_size].copy_(
-                    device_pool.k_buffer[j][d_index : d_index + self.page_size],
-                    non_blocking=True,
-                )
-                self.kv_buffer[1, j, h_index : h_index + self.page_size].copy_(
-                    device_pool.v_buffer[j][d_index : d_index + self.page_size],
-                    non_blocking=True,
-                )
+        transfer_kv_to_cpu_all_layer_naive(
+            host_indices=host_indices,
+            host_k_buffer=self.kv_buffer[0],
+            host_v_buffer=self.kv_buffer[1],
+            device_indices=device_indices,
+            device_k_buffer=device_pool.k_buffer,
+            device_v_buffer=device_pool.v_buffer,
+            page_size=self.page_size,
+            layer_num=self.layer_num,
+        )
 
     def load_page_per_layer(self, host_indices, device_indices, device_pool, layer_id):
-        device_indices_cpu = device_indices[:: self.page_size].cpu()
-        for i in range(len(device_indices_cpu)):
-            h_index = host_indices[i * self.page_size]
-            d_index = device_indices_cpu[i]
-            device_pool.k_buffer[layer_id][d_index : d_index + self.page_size].copy_(
-                self.kv_buffer[0, layer_id, h_index : h_index + self.page_size],
-                non_blocking=True,
-            )
-            device_pool.v_buffer[layer_id][d_index : d_index + self.page_size].copy_(
-                self.kv_buffer[1, layer_id, h_index : h_index + self.page_size],
-                non_blocking=True,
-            )
+        transfer_kv_to_gpu_per_layer_naive(
```

Comment:
The Python part should be included in the subsequent PR after the sgl-kernel releases a new version.